### PR TITLE
DEV-13791: Migrate Example Notebook to use Real Cash Ladder APIs

### PR DESCRIPTION
**Pull request by fox8090**
# Pull Request Checklist

- [x] Changes follow the [style guide](https://github.com/finbourne/sample-notebooks/blob/master/docs/FINBOURNE%20notebook%20style%20guide.ipynb)
- [x] Tests pass
- [x] Raised the PR against the `develop` branch
- [x] Notebook outputs do not contain any priviledged data

# Description of the PR

This updates the `Cash ladder.ipynb` notebook to use the `GetBucketedCashFlows` API. This produces the same data as it is at the moment (it currently uses `GetHoldings`/`GetTransactions` and creates the cash ladder manually).

A new csv datafile has been added to support this containing fictional quotes for various equity instruments. The file `cash_ladder.py` has been removed which contained functions to build the cash ladder manually, which is no longer necessary.

### CODE REVIEW

File: examples/use-cases/cash-management/Cash ladder.ipynb [hunk 1]
```diff
@@ -3,36 +3,37 @@
   {
```
Comment (cscialino) [inline examples/use-cases/cash-management/Cash ladder.ipynb:3]: <pre class="ql-syntax" spellcheck="false">Line #53.        display(ladder)
</pre><p>Very nice -- I wonder if for visualization purposes we could use a MultiIndex DF here that has the date as a 2nd level?</p><p><a href="https://pandas.pydata.org/docs/reference/api/pandas.MultiIndex.html" target="_blank">pandas.MultiIndex — pandas 2.1.0 documentation (pydata.org)</a></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/738/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

Comment (jyaffe) [inline examples/use-cases/cash-management/Cash ladder.ipynb:3]: <pre class="ql-syntax" spellcheck="false">Line #14.            rounding_method="RoundDown",
</pre><p>Is it possible to explain what this rounding_method is doing and what other options are? Can link to a support doc that covers it if one exists. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/738/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

Comment (jyaffe) [inline examples/use-cases/cash-management/Cash ladder.ipynb:3]: <p>Can you rephrase this a little to say that we can construct and view the cash ladder in multiple currencies, e.g. portfolio base currency and any other currency with available quote data</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/738/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='27'/>

Comment (fox8090) [reply examples/use-cases/cash-management/Cash ladder.ipynb:3]: Good point. I've expanded the comment for clarification.

Comment (fox8090) [reply examples/use-cases/cash-management/Cash ladder.ipynb:3]: Good call. I've altered the outputted tables to improve their readability.

Comment (jyaffe) [inline examples/use-cases/cash-management/Cash ladder.ipynb:3]: <pre class="ql-syntax" spellcheck="false">Line #9.    cash
</pre><p>Can you add <code>bucketed cashflow</code> and <code>cash ladder</code> into this list alongside <code>cash</code>?     It helps build out the index on <a href="https://github.com/finbourne/sample-notebooks/tree/master/examples" target="_blank">https://github.com/finbourne/sample-notebooks/tree/master/examples</a></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/738/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (jyaffe) [inline examples/use-cases/cash-management/Cash ladder.ipynb:3]: <p>typo</p><blockquote>we will set upsert a portfolio</blockquote><p> </p><p> </p><p> </p><p>rephrase second sentence:</p><blockquote>Then we will use the GetBucketedCashFlows API to build a cash ladder to understand how those transactions affect our balance over a range of dates. </blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/738/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (jyaffe) [inline examples/use-cases/cash-management/Cash ladder.ipynb:3]: <pre class="ql-syntax" spellcheck="false">Line #18.        scope,
</pre><p>I know this isn't your change, but please can you add the parameter names into this code? Like L21. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/738/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (fox8090) [reply examples/use-cases/cash-management/Cash ladder.ipynb:3]: Thanks for spotting these

**Review by Seamus-Salt — APPROVED** [review]
/LGTM

**Review by jyaffe — APPROVED** [review]
